pkg/cluster/mpi/statusupdate.go

```diff
@@ -44,44 +47,39 @@ func (c statusUpdateComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
 		modified = true
 	}
 
-	// update node list
-	podList := &corev1.PodList{}
-	listOpts := []client.ListOption{
-		client.InNamespace(cr.Namespace),
-		client.MatchingLabels(meta.StandardLabels(cr)),
+	pods, err := getPods(ctx, cr)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ctrl.Result{}, fmt.Errorf("cannot list cluster pods: %w", err)
 	}
-	if lErr := ctx.Client.List(ctx, podList, listOpts...); lErr != nil {
-		return ctrl.Result{}, fmt.Errorf("cannot list cluster pods: %w", lErr)
-	}
-
 	var podNames []string
-	for _, pod := range podList.Items {
+	var runningPodCnt = 0
+	for _, pod := range pods {
 		podNames = append(podNames, pod.Name)
+		if pod.Status.Phase == corev1.PodRunning {
```

Comment:
In the current nucleus code, we only treat the master as Ready if phase is ready and the pod condition ready is also true (I believe a pod goes to phase ready as soon as it is scheduled but only to the pod condition when it is up and healthy) or something like that

Is the choice to use only phase and not condition intentional?

Reply:
That's a good point. I didn't use conditions because line 58 above is already an improvement over the previous logic, and I though checking something else will be an overkill. After Pod is scheduled it goes to Pending first, and only after that to Ready. So, the current logic should be _almost_ OK. But I will add checking for conditions too. 

Q: Does Nucleus check for conditions on Pod or Container? (I believe it's possible to do either way).

Reply:
IMO we should determine what to check to confirm that the readiness / liveness checks are passing for the Pod. I believe that might be checking that the Pods "Ready" condition is `true`. https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

Reply:
I agree. I think you both (@JoshBroomberg, @ddl-kevin) are talking about the same thing. It's an easy change.